### DISCUSSION TITLE
multipart/formdata parser

### DISCUSSION THREAD

**Discussion by kapouer** [nodejs/help#4627] [Ideas]
Dealing with multipart/formdata parsing is something that is very close to how Node manage streams, and doing it right, without introducing huge security vulnerabilities, is somewhat a difficult task.
Result: everyone use busboy, (a not-so-well-maintained module, IMO full of anti-patterns).
It would improve a lot "being able to upload to a Node web server" task to have a parser in the http module.

**Comment by mateuspaulino** [comment]
Hey @kapouer, have you tried cloudnary? I've used it a few times and had no problem with support so far
https://cloudinary.com/documentation/node_integration

**Comment by kapouer** [comment]
Hi Mateus,
I already *have* a solution for that problem in particular.
My idea is: that kind of solution should be implemented in *core* nodejs because it is a *standard*.

**Reply by mateuspaulino** [reply]
Great!

**Comment by Tommy1223454** [comment]
When testing multipart routes with `inject`, I’d treat real upload fixtures as first-class test inputs instead of only checking transport success.

The interesting test cases are usually:
- expected valid file
- wrong extension / mismatched content
- archive edge cases if ZIPs are accepted
- known-bad fixture files

I built an OSS package for this inspection step in Node.js:
https://github.com/pompelmi/pompelmi

Even if your transport test is already passing, adding “untrusted upload” fixtures usually gives much better coverage for real-world file handling.
